### PR TITLE
Thoughts on CBD

### PR DESCRIPTION
**Pull request by jpdrecourt**
A first version of CBD, rather terse. Let me know what you think.

### CODE REVIEW

File: src/thoughts-on/contract-based-web-design.html [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  <body class="body">
+    <header>
+      <h1>
+        Contract-based Web Design: From Protocols to Natural Language Contracts
+      </h1>
+    </header>
+    <main class="main">
+      <p>
+        Within the next decade, automated AI agents will mediate most web
+        browsing. In such a paradigm, the owners of content will have less or no
```
Comment (danhartley) [inline src/thoughts-on/contract-based-web-design.html:19]: little or no

File: src/thoughts-on/contract-based-web-design.html [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      </p>
+      <ul>
+        <li>
+          Observability: Established during handshake between consumer and data
+          owner
+        </li>
+        <li>
+          Verifiability: Using blockchain or repositories for verification
+        </li>
+        <li>Privity: Knowledge distribution limited to necessary parties</li>
```
Comment (danhartley) [inline src/thoughts-on/contract-based-web-design.html:41]: Is privity necessary or would privacy do?

File: src/thoughts-on/contract-based-web-design.html [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      </p>
+
+      <h2>Verification and Security</h2>
+
+      <p>
+        While embracing natural language flexibility, these contracts maintain
+        security through:
+      </p>
+      <ul>
+        <li>Blockchain resilience via decentralized networks</li>
```
Comment (danhartley) [inline src/thoughts-on/contract-based-web-design.html:98]: Usually decentralised in GB English

**Review by danhartley — APPROVED** [review]
It's good! I think it's fine as is, it sets out the principal idea well. Implementation is for another day (or 10).

**Comment by jpdrecourt** [comment]
Great thanks, I've applied the corrections and added an extra explanation about privity. 

As for implementation... we'll need a few more pains aux raisins!

File: src/thoughts-on/contract-based-web-design.html (hunk 1 above)
Comment (danhartley) [reply src/thoughts-on/contract-based-web-design.html]: Great. I like the Key Concepts too.

**Pull request merged**